lua/SS13.lua

```diff
@@ -0,0 +1,79 @@
+local SS13 = {}
+
+SS13.SSlua = dm.global_vars:get_var("SSlua")
+
+SS13.global_proc = "some_magic_bullshit"
+
+local states = SS13.SSlua:get_var("states"):to_table()
+for _, state in pairs(states) do
+	if state:get_var("internal_id") == dm.state_id then
+		SS13.state = state
+		break
+	end
+end
+
+function SS13.new(type, ...)
+	local args = table.pack(...)
+	args["n"] = nil
+	local datum = dm.global_proc("_new", type, args)
+	local references = SS13.state:get_var("references")
+	references:add(datum)
+	return datum
+end
+
+function SS13.await(thing_to_call, proc_to_call, ...)
+	if thing_to_call == nil then
+		thing_to_call = SS13.global_proc
+	end
+	if thing_to_call == SS13.global_proc then
+		proc_to_call = "/proc/" .. proc_to_call
+	end
+	local args = table.pack(thing_to_call, proc_to_call, ...)
+	args["n"] = nil
+	local promise = SS13.new("/datum/promise", table.unpack(args))
+	while promise:get_var("status") == 0 do
+		sleep()
+	end
+	return promise:get_var("return_value"), promise:get_var("runtime_message")
+end
+
+function SS13.wait(time, _timer)
+	local index = #__yield_table + 1
+	local callback = SS13.new("/datum/callback", SS13.SSlua, "queue_resume", SS13.state, index)
+	local timer = dm.global_proc("_addtimer", callback, time*10, 8, _timer, "lua/SS13.lua", 43)
```

Comment:
What is that magic 43?

Reply:
Yeah, plus it's actually obvious what it is.

Note that `debug.info` shouldn't be used for anything load bearing (it can break when inlining is involved, for instance), but its use case is very fine here.